codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/shapevisitor/DafnyToSmithyShapeVisitor.java

```diff
@@ -171,11 +171,12 @@ protected String referenceStructureShape(final StructureShape shape) {
           );
       }
       if (!this.isOptional) {
-        return "return &%s{%s}".formatted(
+        return "return &%s{%s.(*%s)}".formatted(
```

Comment:
I thought this would get fixed by using concrete types over `interface{}` for Positional. Is it not?